api/src/main/java/org/openmrs/ChemoDosingInstructions.java

```diff
@@ -0,0 +1,171 @@
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can 
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under 
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS 
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs;
+
+import java.util.Date;
+import java.util.Locale;
+import java.util.StringTokenizer;
+
+import org.openmrs.api.APIException;
+import org.springframework.validation.Errors;
+import org.springframework.validation.ValidationUtils;
+
+/**
+ * @since 2.4.0
+ */
+public class ChemoDosingInstructions implements DosingInstructions {
+
+	private Double dosageAdjustmentPercentage;
+
+	private String dosageDeliveredTiming;
+
+	private String dosageDilutionInstructions;
+
+	/**
+	 * @see DosingInstructions#getDosingInstructionsAsString(java.util.Locale)
+	 */
+	@Override
+	public String getDosingInstructionsAsString(Locale locale) {
+		StringBuilder dosingInstructions = new StringBuilder();
+
+		// serializing adjustment percentage, timing and dilution instructions to compose dosing instructions
+		if (this.dosageAdjustmentPercentage != null) {
+			dosingInstructions.append("Dosage Adjustment Percentage : ");
+			dosingInstructions.append(this.dosageAdjustmentPercentage);
+			dosingInstructions.append("\n");
+		}
+		if (this.dosageDeliveredTiming != null) {
+			dosingInstructions.append("Dosage Delivered Timing : ");
+			dosingInstructions.append(this.dosageDeliveredTiming);
+			dosingInstructions.append("\n");
+		}
+		if (this.dosageDilutionInstructions != null) {
+			dosingInstructions.append("Dosage Dilution Instructions : ");
+			dosingInstructions.append(this.dosageDilutionInstructions);
+			dosingInstructions.append("\n");
+		}
+		return dosingInstructions.toString();
+	}
+
+	/**
+	 * @see DosingInstructions#setDosingInstructions(DrugOrder)
+	 */
+	@Override
+	public void setDosingInstructions(DrugOrder order) {
+
+		// storing dosing instruction type as "ChemoDosingInstructions"
+		order.setDosingType(this.getClass());
+
+		// storing dosing instruction string into DrugOrder object
+		order.setDosingInstructions(this.getDosingInstructionsAsString(Locale.getDefault()));
+	}
+
+	/**
+	 * @see DosingInstructions#getDosingInstructions(DrugOrder)
+	 */
+	@Override
+	public DosingInstructions getDosingInstructions(DrugOrder order) {
+		if (!order.getDosingType().equals(this.getClass())) {
+			throw new APIException("DrugOrder.error.dosingTypeIsMismatched", new Object[] { this.getClass().getName(),
+				order.getDosingType() });
+		}
+		ChemoDosingInstructions chemoDosingInstructions = new ChemoDosingInstructions();
+
+		// dosing instructions in drug order consist of several lines each having a different type of instruction
+		StringTokenizer tokenizeDosingInstructions = new StringTokenizer(order.getDosingInstructions(), "\n");
```

Comment:
`StringTokenizer tokenizeDosingInstructions =    new  StringTokenizer(order.getDosingInstructions(), "\n");`
add some space to string tokenizer